tests/ui-fulldeps/stable-mir/check_attribute.rs

```diff
@@ -40,7 +40,7 @@ fn test_stable_mir() -> ControlFlow<()> {
 fn test_builtins(items: &CrateItems) {
     let target_fn = *get_item(&items, "builtins_fn").unwrap();
     let allow_attrs = target_fn.attrs_by_path(&["allow".to_string()]);
-    assert_eq!(allow_attrs[0].as_str(), "#![allow(unused_variables)]");
+    assert_eq!(allow_attrs[0].as_str(), "#[allow(unused_variables)]");
```

Comment:
hir attributes don't necessarily store whether they were inner or outer anymore (this shouldn't be relevant anymore, anything that cares about that should emit diagnostics on the ast instead). However, more generally, `attrs_by_path` isn't really a thing anymore because attributes don't necessarily have a path anymore. I can rewrite this to a `get_attr!()` call, but am a little doubtful how useful this test is in general.

Reply:
I guess I will just fix the test, but I'm not convinced it does much

Reply:
ugh, no smir has its own definition of `Attribute`, not the one from hir, matching on them is impossible and the entire `ttrs_by_path` approach is pretty much obsolete

Reply:
removed smir apis for attributes for now, since keeping it around would expose an interface that for some attributes would just panic until all attributes are converted...